01-node-tutorial/answers/16-streams.js

```diff
@@ -6,4 +6,4 @@ stream.on('data', (result) => {
   console.log(result);
 });
 
-stream.on('error', (err) => console.log(err));
+stream.on('error:', (err) => console.log(err));
```

Comment:
Unfortunately, this won't work correctly anymore with this change.

If we take a look at the [documentation](https://nodejs.org/api/fs.html#fscreatereadstreampath-options) for the `fs.createReadStream` function, we can see that the function returns a `fs.ReadStream` object that has some event emitters already attached to it. One of those events that gets emitted is called `'error'` : [https://nodejs.org/api/stream.html#event-error_1](https://nodejs.org/api/stream.html#event-error_1)

So that means inside the `fs` module in node, someone has already written some code, similar to what you did in exercise 13. Something like `readStream.emit('error', new Error("whatever error actually occured")`. This means that the developer can create event **listeners** that will get called anytime these events are **emitted**, but the listener has to be told what the name of the event is. In this case, the name of the event is `'error'`. 

In your code, you're telling the `stream` variable to listen to emitted events with the name `'error:'` with a colon, but there's no such thing; so if an error does occur, your listener won't be notified.

For example, if you change line 3 to point to a file that doesn't exist : `const stream = createReadStream('./content/fake_file.txt');`, and run this script; you'll notice that node quits with an error (it'll even tell you that an `'error'` event was emitted):

```
node:events:368
      throw er; // Unhandled 'error' event
      ^

Error: ENOENT: no such file or directory, open './content/big.txtxx'
Emitted 'error' event on ReadStream instance at:
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: './content/big.txtxx'
}
```

But if you fix your `stream.on` listener, then instead of quitting, it should handle the error by logging it. You can make it a bit more clear by writing line 9 with something like: ``stream.on('error', (err) => console.log(`Encountered an error while reading the file: ${err}`));``

In which case instead of node quitting it'll log the message like this:

```
Encountered an error while reading the file: Error: ENOENT: no such file or directory, open './content/big.txtxx'
```